File: JavaAlgorithm/src/main/java/algo/tzashinorpu/ThirdRound/Chapter03/moveZeroes_283.java
package algo.tzashinorpu.ThirdRound.Chapter03;

public class moveZeroes_283 {
    public void moveZeroes(int[] nums) {
        // 0,1,0,3,12 -> 1,3,12,0,0
        int pos = 0;
        for (int i = 0; i < nums.length; i++) {
            int tmp = nums[pos];
            if (nums[i] != 0) {
                nums[pos++] = nums[i];
                nums[i] = tmp;
            }
        }
    }
}

File: Clion/labuladong/chapter0/array/moveZeroes_283.cpp
//
// Created by TzashiNorpu on 2023/6/2.
//
#include "../../header/header.h"

using namespace labuladong;

void Solution::moveZeroes(vector<int> &nums) {
  int pos = 0;
  for (int i = 0; i < nums.size(); ++i) {
    if (nums[i] != 0) {
      nums[pos++] = nums[i];
      if (pos - 1 != i)
        nums[i] = 0;
    }
  }
}